rust-backend/Dockerfile

```
@@ -0,0 +1,68 @@
+# Multi-stage Dockerfile for Rust Weather Backend
+
+# Build stage
+FROM rust:1.75-slim as builder
+
+# Install build dependencies
+RUN apt-get update && apt-get install -y \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)